pandas/core/internals/array_manager.py

```diff
@@ -473,15 +509,19 @@ def is_single_block(self) -> bool:
 
     def get_bool_data(self, copy: bool = False) -> ArrayManager:
         """
-        Parameters
-        ----------
-        copy : bool, default False
-            Whether to copy the blocks
+        Select columns that are bool-dtype and object-dtype columns that are all-bool.
         """
-        mask = np.array([is_bool_dtype(t) for t in self.get_dtypes()], dtype="object")
-        arrays = [self.arrays[i] for i in np.nonzero(mask)[0]]
+        arrays = []
+        indices = []
+        for i, arr in enumerate(self.arrays):
+            if is_bool_dtype(arr.dtype) or (
+                is_object_dtype(arr.dtype) and lib.is_bool_array(arr)
+            ):
+                arrays.append(arr)
+                indices.append(i)
```

Comment:
changed behavior; can we add/enable a test?

Reply:
This PR enables all of `pandas/tests/frame/test_reductions.py` and `pandas/tests/reductions/`. Is that enough? ;)

We seem to have some `get_bool_data` specific tests outside of those tests, though. `Eg in pandas/tests/generic/test_generic.py`. Will add this one to the list as well.

I will probably also need to create a variant of `pandas/tests/internals/test_internals.py` (or rewrite parts to make it work for both types of managers), but would prefer to keep that for another PR though.

Reply:
fair enough